src/main/java/org/springframework/data/tarantool/core/TarantoolTemplate.java

```diff
@@ -218,92 +215,120 @@ private TupleOperations setNonNullFieldsFromTuple(TarantoolTuple tuple) {
+                              List<?> parameters,
+                              ValueConverter<Value, T> entityConverter) {
         Assert.hasText(functionName, "Function name must not be null or empty!");
         Assert.notNull(parameters, "Parameters must not be null!");
         Assert.notNull(entityConverter, "Entity converter must not be null!");
 
-        List<T> result = callForList(functionName, parameters, entityConverter);
+        List<T> result = callForTupleList(functionName, parameters, entityConverter);
         return result != null && result.size() > 0 ? result.get(0) : null;
     }
 
     @Override
-    public <T> T call(String functionName, Class<T> entityType) {
-        return call(functionName, Collections.emptyList(), entityType);
+    public <T> T callForTuple(String functionName, Class<T> entityType) {
+        return callForTuple(functionName, Collections.emptyList(), entityType);
     }
 
     @Override
-    public <T> T call(String functionName, ValueConverter<Value, T> entityConverter) {
-        return call(functionName, Collections.emptyList(), entityConverter);
+    public <T> T callForTuple(String functionName, ValueConverter<Value, T> entityConverter) {
+        return callForTuple(functionName, Collections.emptyList(), entityConverter);
     }
 
     @Override
-    public <T> List<T> callForList(String functionName, Object[] parameters, Class<T> entityClass) {
-        return callForList(functionName, Arrays.asList(parameters), entityClass);
+    public <T> List<T> callForTupleList(String functionName, Object[] parameters, Class<T> entityClass) {
+        return callForTupleList(functionName, Arrays.asList(parameters), entityClass);
     }
 
     @Override
-    public <T> List<T> callForList(String functionName,
-                                   Object[] parameters,
-                                   ValueConverter<Value, T> entityConverter) {
-        return callForList(functionName, Arrays.asList(parameters), entityConverter);
+    public <T> List<T> callForTupleList(String functionName,
+                                        Object[] parameters,
+                                        ValueConverter<Value, T> entityConverter) {
+        return callForTupleList(functionName, Arrays.asList(parameters), entityConverter);
     }
 
     @Override
-    public <T> List<T> callForList(String functionName, List<?> parameters, Class<T> entityClass) {
+    public <T> List<T> callForTupleList(String functionName, List<?> parameters, Class<T> entityClass) {
         Assert.hasText(functionName, "Function name must not be null or empty!");
         Assert.notNull(parameters, "Parameters must not be null!");
         Assert.notNull(entityClass, "Entity class must not be null!");
 
         return executeSync(getResultSupplier(
-                functionName, parameters, tarantoolClient.getConfig().getMessagePackMapper(), entityClass)
+                functionName, parameters, getMessagePackMapper(), entityClass)
```

Comment:
Probably, `getResultSupplier` needs refactoring and merging with the new method cases. Take a look at it